builder/transaction_builder.go

```diff
@@ -146,42 +146,62 @@ func (b *TransactionBuilder) AllotRequiredManaAndStoreRemainingManaInOutput(targ
 		return b
 	}
 
-	minManalockedSlot := b.transaction.CreationSlot + 2*b.api.ProtocolParameters().MaxCommittableAge()
+	if storedManaOutputIndex >= len(b.transaction.Outputs) {
+		return setBuildError(ierrors.Errorf("given storedManaOutputIndex does not exist: %d", storedManaOutputIndex))
+	}
 
-	// check if the output is locked for a certain time to an account.
-	hasManalockCondition := func(output iotago.Output) (iotago.AccountID, bool) {
-		if !output.UnlockConditionSet().HasTimelockUntil(minManalockedSlot) {
-			return iotago.EmptyAccountID, false
-		}
+	// calculate the minimum required mana to issue the block
+	minRequiredMana, err := b.MinRequiredAllotedMana(b.api.ProtocolParameters().WorkScoreStructure(), rmc, blockIssuerAccountID)
+	if err != nil {
+		return setBuildError(ierrors.Wrap(err, "failed to calculate the minimum required mana to issue the block"))
+	}
 
-		unlockAddress := output.UnlockConditionSet().Address()
-		if unlockAddress == nil {
-			return iotago.EmptyAccountID, false
-		}
+	unboundManaInputsLeftoverBalance, err := b.calculateMaximumPossibleAllotment(targetSlot, minRequiredMana, blockIssuerAccountID)
```

Comment:
I find this function name confusing. Why is this about the maximum possible allotment, and not the leftover available mana?